lib/puppet/parser/resource.rb

```diff
@@ -74,15 +74,12 @@ def add_edge_to_stage
   def evaluate
     return if evaluated?
     @evaluated = true
-    if klass = resource_type and ! builtin_type?
-      finish
-      evaluated_code = klass.evaluate_code(self)
```

Comment:
I can't figure out what happened with evaluation after this change and how the evaluate method can set @evaluated to true when it does not actually evaluate